### PR TITLE
nvme: add uri support

### PR DESCRIPTION
**Pull request by gila**
Add parsing of nvmf uri's to the admin library.

@mtzaurus will use this to move the tests that use NBD to NVMe

### CODE REVIEW

File: nvmeadm/src/error.rs [hunk 1]
```diff
@@ -44,6 +44,10 @@ pub enum NvmeError {
         source: glob::PatternError,
         path_prefix: String,
     },
+    #[snafu(display("NVMe URI invalid: {}", source))]
+    UrlError { source: url::ParseError },
+    #[snafu(display("Transported type {} not supported", trtype))]
```
Comment (jonathan-teh) [inline nvmeadm/src/error.rs:49]: Typo: Transport type.

**Review by jonathan-teh — APPROVED** [review]

**Review by blaisedias — APPROVED** [review]
LGTM

**Comment by gila** [comment]
bors merge

**Comment by gila** [comment]
bors merge

**Pull request merged**